File: 1_introduction/1.4.3/e_1_statistics.py
# Statistics - Kattis

from typing import List

def solve(line: List):
    min_value = min(line)
    max_value = max(line)
    range_value = max_value - min_value
    return min_value, max_value, range_value

i = 1
while True:
    try:
        line = input()
    except EOFError:
        break
    line = [int(x) for x in line.strip().split(" ")]
    
    mn, mx, rg = solve(line[1:])
    print(f"Case {i}: {mn} {mx} {rg}")
    i += 1

File: 1_introduction/1.4.3/e_1_statistics.js
// Statistics

function main() {
    let line = readline()
    let i = 1
    while (line !== undefined) {
        let [_, ...data] = line.split(" ").map(n => parseInt(n))
        let min = Math.min(...data)
        let max = Math.max(...data)
        let range = max - min
        console.log(`Case ${i}: ${min} ${max} ${range}`)
        line = readline()
        i++
    }
}

// Readline

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readline() {
    // returns empty string ("") if no input is given.
    // returns undefined when array overflow happens.
    return inputString[currentLine++];
}